dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/event/RumEventSourceResolverTest.kt

```diff
@@ -0,0 +1,331 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.domain.event
+
+import android.util.Log
+import com.datadog.android.log.internal.logger.LogHandler
+import com.datadog.android.rum.model.ActionEvent
+import com.datadog.android.rum.model.ErrorEvent
+import com.datadog.android.rum.model.LongTaskEvent
+import com.datadog.android.rum.model.ResourceEvent
+import com.datadog.android.rum.model.ViewEvent
+import com.datadog.android.utils.forge.Configurator
+import com.datadog.android.utils.forge.aStringNotMatchingSet
+import com.datadog.android.utils.mockDevLogHandler
+import com.nhaarman.mockitokotlin2.argThat
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.verify
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import java.util.Locale
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class RumEventSourceResolverTest {
+
+    lateinit var testedRumEventSourceResolver: RumEventSourceResolver
+
+    lateinit var mockDevLoggerHandler: LogHandler
+
+    @BeforeEach
+    fun `set up`() {
+        testedRumEventSourceResolver = RumEventSourceResolver()
+        mockDevLoggerHandler = mockDevLogHandler()
+    }
+
+    // region View
+
+    @Test
+    fun `M resolve the ViewEvent source W resolveViewEventSource`(forge: Forge) {
+        // Given
+        val fakeSourceName = forge.aValueFrom(ViewEvent.Source::class.java).toJson().asString
```

Comment:
Instead of this it'd be better to make 
```
    forge.anElementFrom(knownSources)

// …

const val knownSources = arrayOf("android", "browser", "flutter", "ios", "react-native")
```

Reply:
The reason why I prefer that way is that if we later introduce a new source in the schema we will definitely forget to update these tests (adding it to the `knownSources`)

Reply:
ok